tests/test_date.py

```diff
@@ -132,76 +205,180 @@ def test_periods(self, low, high, period, expected_results):
         'some_period',
     ])
     def test_should_reject_easily_mistaken_dateutil_arguments(self, invalid_period):
-        low = datetime(2014, 6, 15)
-        period = datetime(2014, 6, 25)
-
-        with self.assertRaisesRegexp(ValueError, "Invalid period: {}".format(invalid_period)):
-            date.get_intersecting_periods(low, period, period=invalid_period).next()
+        self.given_period_low(datetime(2014, 6, 15))
+        self.given_period(datetime(2014, 6, 25))
+        self.given_invalid_period(invalid_period)
+        self.then_should_reject_easily_mistaken_dateutil_arguments()
 
     @parameterized.expand([
-        param(low=datetime(2014, 4, 15), high=datetime(2014, 4, 14)),
-        param(low=datetime(2014, 4, 25), high=datetime(2014, 4, 25)),
+        param(low=datetime(2014, 4, 15), high=datetime(2014, 4, 14), period='month'),
+        param(low=datetime(2014, 4, 25), high=datetime(2014, 4, 25), period='month'),
     ])
-    def test_empty_period(self, low, high):
-        results = list(date.get_intersecting_periods(low, high, period='month'))
-        self.assertEquals([], results)
+    def test_empty_period(self, low, high, period):
+        self.given_period_low(low)
+        self.given_period_high(high)
+        self.given_period(period)
+        self.when_empty_period_is_parsed()
+        self.then_period_is_empty()
+
+    def given_expected_results(self, expected_results):
+        self.expected_results = expected_results
+
+    def given_dates(self, dates):
+        self.dates = dates
+
+    def given_period_low(self, low):
+        self.period_low = low
 
+    def given_invalid_period(self, invalid_period):
+        self.invalid_period = invalid_period
 
-class ParseDateWithFormats(unittest.TestCase):
+    def given_period_high(self, high):
+        self.period_high = high
 
-    def test_shouldnt_parse_invalid_date(self):
-        self.assertIsNone(date.parse_with_formats('yesterday', ['%Y-%m-%d'])['date_obj'])
+    def given_period(self, period):
+        self.period = period
+
+    def when_date_intersecting_generated(self):
+        self.intersected_dates = self.dates
```

Comment:
`when_` methods are actions that we are trying to test. What are we testing here then?